.github/workflows/build-lint-test.yml

```diff
@@ -18,64 +45,135 @@ jobs:
         uses: actions/setup-node@v3
         with:
           node-version: ${{ matrix.node-version }}
-      - name: Get Yarn version
-        run: echo "YARN_VERSION=$(yarn --version)" >> $GITHUB_OUTPUT
-        id: yarn-version
-      - name: Cache yarn dependencies
+      - name: Restore Yarn dependencies
         uses: actions/cache@v3
         with:
-          path: .yarn/cache
-          key: yarn-cache-${{ runner.os }}-${{ steps.yarn-version.outputs.YARN_VERSION }}-${{ hashFiles('yarn.lock') }}
-      - run: yarn install --immutable
+          path: ${{ needs.prepare.outputs.YARN_CACHE_DIR }}
+          key: yarn-cache-${{ runner.os }}-${{ needs.prepare.outputs.YARN_VERSION }}-${{ hashFiles('yarn.lock') }}
+      - run: yarn --immutable
       - run: yarn build
+      - run: yarn workspace @metamask/snap-controllers run test:ci
```

Comment:
There might be a better way to approach this. `snap-controllers` needs the Webpack bundles from `execution-envrionments` to run tests, so rather than running it with the other tests, it's run here.

Reply:
Yeah, that doesn't seem ideal 🤔

Reply:
Lets create an issue to tackle this later